uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -394,35 +440,36 @@ define([
       var request = vm.request;
 
       if (!vm.isMode('create') && isCalculationUnit('hours')) {
-        _.each(['from', 'to'], function (type) {
-          times[type].skipValueSetting = true;
-          times[type].time = extractTimeFromServerDate(request[type + '_date']);
-          times[type].amount = request[type + '_date_amount'];
-          times[type].maxAmount = times[type].amount;
-        });
+        return $q.all(['from', 'to'].map(function (type) {
+          return loadTimeRangesFromWorkPattern(type)
+            .then(function () {
+              times[type].time = extractTimeFromServerDate(request[type + '_date']);
+              times[type].amount = Math.min(request[type + '_date_amount'], times[type].maxAmount).toString();
+            });
+        })).then(setRequestHoursDeductions);
       }
     }
 
     /**
      * Initialises watchers for time and date inputs
      */
     function initTimeAndDateInputsWatchers () {
-      if (vm.isMode('view') || isLeaveType('toil')) { return; }
+      if (vm.isMode('view') || isLeaveType('toil')) {
```

Comment:
remove this condition from here, and use it externally to decide whether to call this function or not

Reply:
Done.